bindgen/templates/CallbackInterfaceImpl.cs

```diff
@@ -42,9 +43,78 @@ class {{ callback_impl_name }} {
                 }
             }
             {%- when None %}
-
             {%- endmatch %}
 
+            {%- else %}
+            CancellationTokenSource cts = new CancellationTokenSource();
+
+            Task.Run(async () => {
+                var ret = new _UniFFILib.{{ meth.foreign_future_ffi_result_struct().name()|ffi_struct_name }}();
+                ret.@callStatus = new UniffiRustCallStatus();
+
+                {%- match meth.throws_type() %}
+                {%- when Some with (error_type) %}
+                try {
+                {%- when None %}
+                {%- endmatch %}
+
+                {%- match meth.return_type() %}
+                {%- when Some with (return_type) %}
+                var result =
+                {%- when None %}
+                {%- endmatch %}
+
+                #if NET6_0_OR_GREATER
+                await uniffiObject.{{ meth.name()|fn_name }}(
+                    {%- for arg in meth.arguments() %}
+                    {{ arg|lift_fn }}({{ arg.name()|var_name }}){%- if !loop.last %}, {% endif -%}
+                    {%- endfor %}).WaitAsync(cts.Token);
+                #else
+                await uniffiObject.{{ meth.name()|fn_name }}(
+                    {%- for arg in meth.arguments() %}
+                    {{ arg|lift_fn }}({{ arg.name()|var_name }}){%- if !loop.last %}, {% endif -%}
+                    {%- endfor %});
+                #endif
```

Comment:
```suggestion
                await uniffiObject.{{ meth.name()|fn_name }}(
                    {%- for arg in meth.arguments() %}
                    {{ arg|lift_fn }}({{ arg.name()|var_name }}){%- if !loop.last %}, {% endif -%}
                    {%- endfor %})
                    #if NET6_0_OR_GREATER
                    .WaitAsync(cts.Token)
                    #endif
                    ;
```
Maybe something along these lines would be a little more compact and cleaner?